Fixes more cases of R -> Turbo A outside dialogue

miriamlefae opened the pull request:
Fixes more cases of R acting as Turbo A when not in overworld dialogue following up on #84

## Description
- Fix more cases of r -> turbo when it shouldn't do be.
- Add two helper functions to clear and set the flag.
- Fixed field_region_map, wallclock and pc.

## **Discord contact info**
kildemal

miriamlefae committed:
fix: r -> turbo a interaction (contd.)

- Fix more cases of r -> turbo when it shouldn't do be.
- Add two helper functions to clear and set the flag.
- Fixed field_region_map, wallclock and pc.

iriv24 commented:
why are the helper functions necessary here?
path 1:
`if(x is false)`
   `set x to true`
 
 is the same as just
 path 2:
`set x to true`
 
 is it not?
 
 lets say x is true to start
 path 1:
 `x is true`
 `if(x is false)` -- condition not met
 `x is true`
 
 path 2:
 `x is true`
 `set x to true`
 `x is true`
 
 now lets say x is false to start
 path 1:
 `x is false`
 `if(x is false)` -- condition is met
 `  set x to true`
 `x is true`
 
 path 2:
 `x is false`
 `set x to true`
 `x is true`
 
 same outcome for both possible values of x for both paths, so why have the unnecessary conditional?

kildemal commented:
Does the compiler optimize away the conditional? I suppose even if it isn't this is a micro optimization at best.

Now that I think about it I have no idea if this might be slower. If the conditional is more expensive than setting the value I'm stupid.

iriv24 commented:
> Does the compiler optimize away the conditional? I suppose even if it isn't this is a micro optimization at best.
> 
> Now that I think about it I have no idea if this might be slower. If the conditional is more expensive than setting the value I'm stupid.

not sure if the compiler would optimize it out and even if not its not really a performance hit, i was just asking because it seems silly lol

kildemal commented:
> > Does the compiler optimize away the conditional? I suppose even if it isn't this is a micro optimization at best.
> > Now that I think about it I have no idea if this might be slower. If the conditional is more expensive than setting the value I'm stupid.
> 
> not sure if the compiler would optimize it out and even if not its not really a performance hit, i was just asking because it seems silly lol

That makes sense. I'll remove the checks. But I think wrapping it in a helper function instead of setting the flag directly is slightly more helpful should debugging be needed. Perhaps condense into a single helper that takes a bool?

iriv24 commented:
> > > Does the compiler optimize away the conditional? I suppose even if it isn't this is a micro optimization at best.
> > > Now that I think about it I have no idea if this might be slower. If the conditional is more expensive than setting the value I'm stupid.
> > 
> > 
> > not sure if the compiler would optimize it out and even if not its not really a performance hit, i was just asking because it seems silly lol
> 
> That makes sense. I'll remove the checks. But I think wrapping it in a helper function instead of setting the flag directly is slightly more helpful should debugging be needed. Perhaps condense into a single helper that takes a bool?

sure thats fine with me, i dont have a super strong opinion. my original question was just because the code looked a bit odd to me, you can do whatever you think will be easiest for debugging in the future. performance is not a concern here

miriamlefae committed:
refactor: simplify dialog flag

- Uses one helper function to set the dialog flag and clear it.
- Removed the function wrapper for drawing the frame.

kildemal commented:
> > > > Does the compiler optimize away the conditional? I suppose even if it isn't this is a micro optimization at best.
> > > > Now that I think about it I have no idea if this might be slower. If the conditional is more expensive than setting the value I'm stupid.
> > > 
> > > 
> > > not sure if the compiler would optimize it out and even if not its not really a performance hit, i was just asking because it seems silly lol
> > 
> > 
> > That makes sense. I'll remove the checks. But I think wrapping it in a helper function instead of setting the flag directly is slightly more helpful should debugging be needed. Perhaps condense into a single helper that takes a bool?
> 
> sure thats fine with me, i dont have a super strong opinion. my original question was just because the code looked a bit odd to me, you can do whatever you think will be easiest for debugging in the future. performance is not a concern here

Simplified the implementation. I'm setting the flag to false wherever needed instead of going out of my way to prevent setting it to true when not needed since there are no performance concerns and this is an inexpensive operation. I retained a single helper function that takes a bool in case debug prints need to be added in the future.
